pkg/queue/error.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2022 Outreach Corporation. All Rights Reserved.
+//
+// Description: error
+package queue
+
+import (
+	"fmt"
+)
+
+// NewMaxCapacityError
+func NewMaxCapacityError(capacity int) *MaxCapacityError {
+	return &MaxCapacityError{
+		capacity: capacity,
+	}
+}
+
+var _ error = new(MaxCapacityError)
```

Comment:
Missing comment here

```suggestion
// _ ensures that MaxCapactityError implements the error interface
var _ error = new(MaxCapacityError)
```